From f433ea670f0da0eeed24a035a3f4f355d4a64666 Mon Sep 17 00:00:00 2001
From: Sidhanta Sahu <quic_sidhanta@quicinc.com>
Date: Thu, 3 Oct 2024 14:23:07 -0700
Subject: [PATCH] wifi: mac80211: Clear sdata running flag

During wifi down operation, sdata running flag is not cleared which
could cause an issue during recovery. During ieee80211 reconfiguration,
sdata is not expected to be reconfigured during wifi down operation.
As ieee80211 reconfig is scheduled after recovery of a target, this work
won't be aware of any wifi down operation, so it relies on flag to check
if a sdata is required to be reconfigured. The issue is the flag is not
clear then it will have an scenario where one link is going down and
another could be going up for the same sdata. It is expected to honor
wifi down operation when a down operation is requested.

Fix by clearing the running flag during an interface down operation.

Patch-dependency: None
Patch-work: None

Signed-off-by: Sidhanta Sahu <quic_sidhanta@quicinc.com>
---
 net/mac80211/cfg.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/mac80211/cfg.c b/net/mac80211/cfg.c
index 643e4d8..29c33e4 100644
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -1840,6 +1840,7 @@ static int ieee80211_stop_ap(struct wiph
 	link_conf = link->conf;
 	sdata_assert_lock(sdata);
 
+	clear_bit(SDATA_STATE_RUNNING, &sdata->state);
 	old_beacon = sdata_dereference(link->u.ap.beacon, sdata);
 	if (!old_beacon)
 		return -ENOENT;
-- 
2.34.1

